chaincode/test/AssetManager.ts

```diff
@@ -0,0 +1,419 @@
+import { expect } from "chai";
+import { ethers, waffle } from "hardhat";
+import {
+  prepare,
+  deploy,
+  checkBalance,
+  checkContractBalance,
+  getBigNumber,
+  getTxCost,
+} from "./utilities";
+import { range, cloneDeep } from "lodash";
+const { BigNumber } = require("ethers");
+const Constants = require("./utilities/constants");
+
+const ZERO_ETH = ethers.utils.parseEther("0");
+const ONE_ETH = ethers.utils.parseEther("1");
+const TWO_ETH = ethers.utils.parseEther("2");
+const INITIAL_BALANCE_ETH = ethers.utils.parseEther("10000");
+const INITIAL_AUTH_LIMIT = ethers.utils.parseEther("100000");
+const DUMMY_HEX = "0x";
+
+// let snapshotId: string;
+
+describe("AssetManager", function (this) {
+  before(async function (this) {
```

Comment:
Is `this` referring to a context variables passed in by Mocha? It conflicts with `this` internal variable of the function. Also conflicts with the outer variable `this` (which also conflicts with the outer function's internal variable)